### PR TITLE
Store metadata for translation update. Update user statistic queries 

### PR DESCRIPTION
**Pull request by aeng**
https://bugzilla.redhat.com/show_bug.cgi?id=1235070

### CODE REVIEW

File: zanata-model/src/main/java/org/zanata/model/type/TranslationEntityType.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+package org.zanata.model.type;
+
+import lombok.Getter;
+import org.zanata.model.HTextFlowTarget;
+import org.zanata.model.tm.TransMemoryUnit;
+
+import java.io.Serializable;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
```
Comment (huangp) [inline zanata-model/src/main/java/org/zanata/model/type/TranslationEntityType.java]: Can we add javadoc for this class please

File: zanata-model/src/main/java/org/zanata/model/type/TranslationSourceType.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    GWT_EDITOR_ENTRY("GWT"),
+    JS_EDITOR_ENTRY("JS"),
+    API_UPLOAD("API"),
+    WEB_UPLOAD("WEB"),
+    UNKNOWN("UNK");
+
+    public static final Collection<TranslationSourceType> AUTOMATED_ENTRIES;
+
+    static {
+        AUTOMATED_ENTRIES = Collections.unmodifiableCollection(new HashSet(
```
Comment (huangp) [inline zanata-model/src/main/java/org/zanata/model/type/TranslationSourceType.java]: Could use guava Sets.newHashSet(E...) instead

File: zanata-war/src/main/java/org/zanata/rest/service/AsynchronousProcessResourceService.java [hunk 3]
```diff
@@ -156,6 +157,19 @@ public ProcessStatus startSourceDocCreationOrUpdate(final String idNoSlash,
                                                     // progress
     }
 
+    /**
+     * This is used by functional test ZanataRestCaller for testing purposes.
```
Comment (huangp) [inline zanata-war/src/main/java/org/zanata/rest/service/AsynchronousProcessResourceService.java]: Is this only used by functional test? If yes we need to put it in zanata-test-war module. But I think this is used by client as well.

Comment (aeng) [inline zanata-war/src/main/java/org/zanata/rest/service/AsynchronousProcessResourceService.java]: Double checked, its used by client too.

**Comment by huangp** [comment]
:+1:  except adding more javadoc to a few of the new classes I am okay with the review

**Comment by djansen-redhat** [comment]
Test:
- [ ] Contribution is counted when user translates to fuzzy
- [ ] Contribution is counted when user translates to translated
- [ ] Contribution is counted when user translates to translated, from fuzzy, without change
- [ ] Contribution is counted when user translates to Approved
- [ ] Contribution is not counted when user runs Copy Version to a new version
- [ ] Contribution is not counted when user runs CopyTrans on a version
- [ ] Contribution is not counted when user runs Merge Version on a version
- [ ] Contribution is not counted when user runs TMMerge from editor on a version
- [ ] CopyTrans etc action is visible in editor text flow comments

**Comment by djansen-redhat** [comment]
It appears that history is being copied as-is, and the user is receiving credit for the history events.
We should mark all history that is copied as "automatedEntry" as well.

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested.

**Comment by aeng** [comment]
Merged into integration/master. https://github.com/zanata/zanata-server/commit/70e87501e4b0bdbc8c03be446e384184dfa61285

**Pull request closed without merging**